lib/freebsd.ex

```diff
@@ -36,6 +38,78 @@ defmodule FreeBSD do
 
   defp freebsd_config, do: Mix.Project.config() |> Keyword.fetch!(:freebsd)
 
+  defp freebsd_config(key),
+    do: Mix.Project.config() |> Keyword.fetch!(:freebsd) |> Map.get(key)
+
   defp with_deps(manifest, nil), do: manifest
   defp with_deps(manifest, deps), do: Map.put(manifest, :deps, deps)
+
+  defp with_user(manifest, nil), do: manifest
+  defp with_user(manifest, username), do: Map.put(manifest, :users, [username])
+
+  def with_scripts(manifest, username) do
+    manifest
+    |> Map.put(:scripts, %{
+      "pre-install" =>
+        Enum.join(
+          [
+            create_user_script(username),
+            create_config_dir_script()
+          ],
+          "\n"
+        )
+    })
+  end
+
+  defp create_user_script(nil), do: ""
+
+  defp create_user_script(username) do
+    pkg_name = pkg_name()
+
+    """
+    if [ -n "${PKG_ROOTDIR}" ] && [ "${PKG_ROOTDIR}" != "/" ]; then
+      PW="/usr/sbin/pw -R ${PKG_ROOTDIR}"
+    else
+      PW=/usr/sbin/pw
+    fi
+    echo "===> Creating groups."
+    if ! ${PW} groupshow #{username} >/dev/null 2>&1; then
+      echo "Creating group '#{username}'."
+      ${PW} groupadd #{username}
+    else
+      echo "Using existing group '#{username}'."
+    fi
+    echo "===> Creating user"
+    if ! ${PW} usershow #{username} >/dev/null 2>&1; then
+      echo "Creating user '#{username}'."
+      ${PW} useradd #{username} -g #{username} -c "#{pkg_name} user" -d /usr/local/libexec/#{pkg_name} -s /bin/sh
+    else
+      echo "Using existing user '#{username}'."
+    fi
+    """
+  end
+
+  defp create_config_dir_script() do
```

Comment:
This should be done as part of installation staging, rather than a post-install script. Any files placed in the install dir will be copied as part of installation, and removed as part of de-installation if they've been modified. [Here's an example](https://github.com/patmaddox/ex_freebsd/blob/main/lib/mix/tasks/freebsd.pkg.ex#L43).

Reply:
Wouldn't that overwrite any configuration files as a part of package upgrades?

Reply:
It should replace files that haven’t been modified. Any that have been modified will be untouched.

I’ll take a closer look at some other packages but I believe that’s how it behaves. That’s how a package can provide a default config that is removed on package install - but if it’s been modified then it leaves it alone.

Reply:
It also looks like it deletes the file on package removal, regardless of whether it's been modified. Maybe there's a flag that can be added to the manifest or something.

Reply:
It looks like there's a `post-install` and `pre-deinstall`, so I think this could be rewritten to execute a script installed into `libexec`. I'll test that out and see what it looks like. The upside is that the manifest could always run the scripts, and those scripts might be noop depending on the packaging configuration.